container/sagemaker/serve.py

```diff
@@ -124,10 +126,20 @@ def _start_tfs(self):
         self._tfs = p
 
     def _start_nginx(self):
+        self._log_version('/usr/sbin/nginx -V', 'nginx version info:')
         p = subprocess.Popen('/usr/sbin/nginx -c /sagemaker/nginx.conf'.split())
         log.info('started nginx (pid: %d)', p.pid)
         self._nginx = p
 
+    def _log_version(self, command, message):
+        try:
+            output = subprocess.check_output(
+                command.split(),
+                stderr=subprocess.STDOUT).decode('utf-8', 'backslashreplace').strip()
+            log.info('{}\n{}'.format(message, output))
+        except subprocess.CalledProcessError:
```

Comment:
An error decoding the output will punch through this except block. Not sure if we should also catch that?

Reply:
python docs indicate decode will not raise if an error handler is set (backslashreplace), so this would only raise if output was None, which shouldn't be possible absent a CalledProcessError